### DISCUSSION TITLE
Terminate feed Mid-Stream

### DISCUSSION THREAD

**Discussion by sitehatchery** [owncast/owncast#1568] [Q&A] [answered]
Is there a way to terminate a feed mid-stream? 

We have a few people scheduled to do livestreams remotely. But we need to be able to terminate a feed if necessary. 

We won't have control over their OBS program because they are remote. 

We've tried to change the stream key (didn't work), but outside of building some sort of trigger to restart the Owncast service without having to log into a command terminal, I'm not sure how we would do this.

Any thoughts on this?

**Comment by jeyemwey** [comment]
We have a similar usecase for the demo stream at broadcast.owncast.online, and we use the [disconnect POST request route](https://owncast.online/api/latest/#tag/Admin/paths/~1api~1admin~1disconnect/post) for that. Maybe that can help you?

**Reply by sitehatchery** [reply]
The issue here is that it does terminate, but OBS reconnects it immediately. Have you found a way around this?

**Reply by gabek** [reply]
You could tell your streamers to set OBS to not reconnect (I'm assuming that's a setting?) or you could change the stream key.

**Reply by sitehatchery** [reply]
True, but the reason for the disconnect feature would be for if someone is violating the terms or there's a scheduling conflict. So we'd want more control over it. I'm hoping for a more direct approach. I may need end up capturing their IP address and then just immediately disconnect the broadcaster if they try to connect again, if the IP address matches in the lookup table.

I'm hoping for a more direct approach though, if someone has had to deal with this already.

**Reply by aaronpk** [reply]
tbh it sounds like you might need to do this at a layer outside of OwnCast. You could configure a separate RTMP server for your multiple users to stream to, and feed a single input into OwnCast. For example, a Magewell h264/HDMI decoder could receive one or more RTMP feeds and output it over HDMI into some other device like an ATEM Mini that can stream to OwnCast. That'd give you full control over what's appearing on the stream and the ability to quickly create new stream keys on the fly and also cut off feeds if you need to.

**Comment by sitehatchery** [comment]
Thank you!!!

**Comment by sitehatchery** [comment, marked as answer]
I did this with a combination of strategies. I added a function to disconnect the stream. The stream disconnects, but OBS continuously tries to reconnect when a stream drops. So I then added the broadcasters IP address (gotten from the API) to the database and threw a 404 error on the site if the user tries to access it. The script also changes the stream key after it disconnects. 

However, WordPress (what it's hosted on) needs to know what the stream key is so that it can connect to it in the API. I think it would be better to have the admin login and API key different from the stream key. But in any case, I've added a function to change the stream key from within WordPress so that WordPress knows what it is changed to.
